packages/fxa-auth-server/lib/metrics/events.js

```diff
@@ -104,8 +104,11 @@ module.exports = (log, config) => {
       // eslint-disable-next-line @typescript-eslint/no-this-alias
       const request = this;
 
-      if (data && data.uid && request.auth && request.auth.artifacts) {
-        request.auth.artifacts.metricsUid = data.uid;
+      if (data && data.uid && request.auth) {
+        request.auth.artifacts = {
+          ...request.auth.artifacts,
+          metricsUid: data.uid,
+        };
```

Comment:
Thanks Reino! I can see how this would work, though in the case of the new `fxa_subscribe - subscription_ended` event, the `request` is from Stripe, which [doesn't have an authentication strategy](https://github.com/mozilla/fxa/blob/168e46efcef7baff228e33bcf60268d449fcfa4a/packages/fxa-auth-server/lib/routes/subscriptions/stripe-webhook.ts#L1165), and even if it did, we don't get the `uid` here from the `request` or any `auth` on the `request`.

Is there another place on the `request` we can stash the `uid` that we get from a DB look-up? Then maybe we can update `isMetricsEnabled` to check there?

If not, then at least perhaps we can update the [comment](https://github.com/mozilla/fxa/blob/7fdfe181638435144ee17f1389d27cf90b1d4c6b/packages/fxa-auth-server/lib/server.js#L280-L281) in the `isMetricsEnabled` request decorator to say we also get the uid this way for Stripe webhooks?

Curious if @vbudhram has any thoughts?

Reply:
I'm not super familiar with `hapi` best practices, but from their docs it looks like [request.app](https://hapi.dev/api/?v=21.3.2#-requestapp) could be used for this parameter?

To that end, I've put up [another PR](https://github.com/mozilla/fxa/pull/15298) updating `request.app` instead of `request.auth.artifacts`.

@vbudhram and @biancadanforth any thoughts on the two options? Happy to make any change necessary.

Reply:
No problem at all. Thank you @vbudhram for looking into it.

Reply:
Thanks I added a comment to the other PR you linked; I was concerned `request.app` was global, but it appears to be request-specific after some local testing.